### PR TITLE
feat(ytmusic): add music#getPlaylist()

### PR DESCRIPTION
**Pull request by patrickkfkan**
## Description

This PR adds `getPlaylist()` to ytmusic.

I have refrained from updating the V2 docs as currently the docs do not have anything relating specifically to ytmusic.

Basic usage:
```
// Get playlist with first batch of items
const pl = await youtube.music.getPlaylist('PLVbEymL-83SyVXXqT7fYX5sEvELvyGjL7');

// pl: {
//   header: {...}
//   items: [...]
// }

// Get related playlists
const related = await pl.getRelated();

// related: [...]

// Get next batch of items
if (pl.has_continuation) {
  const pl2 = await pl.getContinuation();
  
  // pl2: {
  //   items: [...]
  // }

  // (Use pl2.has_continuation / pl2.getContinuation() to get more items...)
}
```

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by LuanRT** [comment]
Awesome, thanks! Can you fix the linting errors? Otherwise everything looks good.

**Comment by patrickkfkan** [comment]
Hi, I've committed the "fix" after running  `npx eslint ./src --fix`. This resolves the initial errors but the workflow now reports the following:

```
Error: src/parser/ytmusic/Playlist.ts(66,53): error TS2367: This condition will always return 'false' since the types 'Text | undefined' and 'string' have no overlap.
```
I am unable to reproduce this error locally when running `npm run build` and `playlist.getRelated()` does return correct data. Do you happen to have any idea? Thanks.

File: src/parser/ytmusic/Playlist.ts [hunk 1]
```diff
@@ -60,22 +59,22 @@ class Playlist {
...
-            return related.contents || [];
-        }
-        else {
-            section_continuation = section_list?.continuation;
-        }
+      const response = await this.#actions.browse(section_continuation, { is_ctoken: true, client: 'YTMUSIC' });
+      const data = Parser.parseResponse(response.data);
+      const section_list = data.continuation_contents?.as(SectionListContinuation);
+      const sections = section_list?.contents?.as(MusicCarouselShelf);
+      const related = sections?.filter((section) => section.header?.title === 'Related playlists')[0];
```
Comment (Wykerd) [inline src/parser/ytmusic/Playlist.ts:66]: Replace this line with `const related = sections?.filter((section) => section.header?.title?.toString() === 'Related playlists')[0];` to fix the error message.

Comment (LuanRT) [reply src/parser/ytmusic/Playlist.ts:66]: > Replace this line with `const related = sections?.filter((section) => section.header?.title?.toString() === 'Related playlists')[0];` to fix the error message.


Hmm.
The weird thing though is that `MusicCarouselShelf`'s header is of type `MusicCarouselShelfBasicHeader`. And that header's title is not a `Text`.

https://github.com/LuanRT/YouTube.js/blob/050addbcccd3aaaec5bfc4bf3474496efdaa3d83/src/parser/classes/MusicCarouselShelfBasicHeader.js#L15

Something's wrong here.

Comment (Wykerd) [reply src/parser/ytmusic/Playlist.ts:66]: Yes, something funky is happening in the Github Actions, since it does compile fine when I pull it locally.

Comment (LuanRT) [reply src/parser/ytmusic/Playlist.ts:66]: Same here, I'm gonna just merge this then.

**Pull request merged**